elastic/kibana
# [Observability Onboarding] Add Firehose flow telemetry

Closes https://github.com/elastic/kibana/issues/190791

## Summary

Adds three custom events for the Firehose flow in addition to the automatically tracked click events. The new events are:
* Flow initialization with either success or error
* Starting to monitor for incoming data
* Data received for one of the AWS indices (separate event for each)

## How to test

* Run Kibana locally or use the deployed serverless instance from this PR
* Open browser dev tools
* Go to Firehose flow in Kibana
* Look for `/kibana-browser` requests in the Network tab
* Make sure there is `observability_onboarding_firehose_initialize` event when the flow is initialized either successfully or with an error
* Focus on some other window to trigger data monitoring and go back to Kibana
* Make sure `observability_onboarding_firehose_started_monitor_data` event was triggered
* Ingest a few documents for different indices (see example requests below) and make sure there are `observability_onboarding_firehose_data_received` events for each index when data was detected

## Data ingest requests

```
POST logs-awsfirehose-default/_doc
{
    "@timestamp": "2024-09-20T09:30:00.000Z",
    "agent": {
      "id": "firehose"
    },
    "aws.kinesis.name": "Elastic-CloudwatchLogs"
}

POST metrics-aws.s3_request-default/_doc
{
    "@timestamp": "2024-09-20T13:30:00.000Z",
    "agent": {
      "id": "firehose"
    },
    "aws.kinesis.name": "all-cloudwatch-logs-to-elastic",
    "aws": {
      "cloudwatch": {
        "namespace": "AWS/S3"
      },
      "exporter": {
        "arn": "arn:aws:cloudwatch:us-west-2:975050175126:metric-stream/Elastic-CloudwatchLogsAndMetricsToFirehose-CloudWatchMetricStream-Nhb4NhzPdL4J"
      }
    }
}

POST metrics-aws.apigateway_metrics-default/_doc
{
    "@timestamp": "2024-09-20T13:31:01.000Z",
    "agent": {
      "type": "firehose"
    },
    "aws": {
      "cloudwatch": {
        "namespace": "AWS/ApiGateway"
      },
      "exporter": {
        "arn": "arn:aws:cloudwatch:us-west-2:975050175126:metric-stream/Elastic-CloudwatchLogsAndMetricsToFirehose-CloudWatchMetricStream-Nhb4NhzPdL4J"
      }
    },
    "cloud": {
      "account": {
        "id": "975050175126"
      },
      "provider": "aws",
      "region": "us-west-2"
    }
}
```
